microcluster/app.go

```diff
@@ -96,21 +97,29 @@ func (m *MicroCluster) Start(apiEndpoints []rest.Endpoint, schemaExtensions map[
 		return fmt.Errorf("Unable to start daemon: %w", err)
 	}
 
+	stop := sync.OnceFunc(func() {
+		logger.Info("Initiate daemon stop sequence")
+		if err := d.Stop(); err != nil {
+			d.ShutdownDoneCh <- err
+			return
+		}
+		d.ShutdownDoneCh <- d.ShutdownCtx.Err()
```

Comment:
This needs a new empty line before it otherwise it'll fail our CI checks (well it would on LXD at least).

Reply:
Wont this mean that a "normal" shutdown will cause a Canceled error to be sent down the `d.ShutdownDoneCh` which will cause the daemon to exit with a non-zero exit code, possibly causing it to be restarted systemd?

Reply:
I'm not really seeing why we need to get the shutdown context error at all? (some comments in this part of the code base would be welcomed).

Reply:
My argument was that if the daemon gets stopped due to the parent context being cancelled it can forward the actual reason why its now performing a stop.

Reply:
TBH I find this "parent context" thing a bit odd here. I guess this is because its a package rather than a daemon itself like LXD. But I think we need to differentiate between the **request** that triggers the shutdown (either a signal being received or the parent context being cancelled) and the **result** of the shutdown, which is what needs to influence the exit code of the program.

Also as an aside, I believe its somewhat an anti-pattern to be setting up signal handlers in an imported package.
Not something I would recommend.

Really the parent process should be setting those up and trigger a call to a .Stop() function or cancelling that parent context, rather than the package doing it itself.

Reply:
> Also as an aside, I believe its somewhat an anti-pattern to be setting up signal handlers in an imported package.
Not something I would recommend.
> 
> Really the parent process should be setting those up and trigger a call to a .Stop() function or cancelling that parent context, rather than the package doing it itself.

+100, and this is exactly what we do and how I bumped into the issue here. I would very much like to have a `DisableDefaultSignalHandlers bool` option in microcluster in the future to completely disable this, but that would have to be carefully setup to avoid breaking backwards compatibility.

This PR, admittedly, does not lead to a perfect world, but solves the problem with minimal changes to microcluster itself.

Reply:
CC @masnax